src/framework/ui/modal/modal.spec.tsx

```diff
@@ -3,210 +3,117 @@ import {
+    expect(modal.props.backdropStyle).toBe(modalPassingProps.backdropStyle);
   });
 
-  it('* modal closes on passed prop', () => {
-    const onCloseModal = jest.fn();
-
-    const component: RenderAPI = render(
-      <Modal
-        visible={true}
-        isBackDropAllowed={true}
-        identifier={MODAL_TEST_IDENTIFIER('1')}
-        onCloseModal={onCloseModal}>
-        <View>
-          <Text>Test1</Text>
-          <Button
-            title='Close Modal'
-            onPress={onCloseModal}
-          />
-        </View>
-      </Modal>,
-    );
-
-
-    const { output: openedOutput } = shallow(component.getByType(Modal));
-    expect(openedOutput).toMatchSnapshot();
+  it('* modal shows/hides properly', () => {
+    const component: RenderAPI = render(<TestScreen/>);
 
-    fireEvent.press(component.getByType(Button));
-    expect(onCloseModal).toHaveBeenCalled();
+    fireEvent.press(component.getByTestId(buttonShowModalTestId));
+    expect(component.getByType(Modal).props.visible).toBe(true);
 
-    const { output: closedOutput } = shallow(component.getByType(Modal));
-    expect(closedOutput).toMatchSnapshot();
+    fireEvent.press(component.getByTestId(buttonHideModalTestId));
+    expect(component.getByType(Modal).props.visible).toBe(false);
   });
 
-  it('* modal component close on backDrop checks', () => {
-    const onCloseModal = jest.fn();
-
+  it('* modal default center placement calculated properly', () => {
+    const { width, height } = Dimensions.get('screen');
+    const modalStyle: StyleType = {
+      width: 200,
+      height: 200,
+    };
     const component: RenderAPI = render(
-      <Modal
-        visible={true}
-        isBackDropAllowed={true}
-        identifier={MODAL_TEST_IDENTIFIER('1')}
-        onCloseModal={onCloseModal}>
-        <View>
-          <Text>
-            Test1
-          </Text>
-        </View>
-      </Modal>,
+      <TestScreen modalStyle={modalStyle}/>,
     );
 
-    const { output: openedOutput } = shallow(component.getByType(Modal));
-    expect(openedOutput).toMatchSnapshot();
-
-    fireEvent.press(component.getByType(TouchableOpacity));
-    expect(onCloseModal).toHaveBeenCalled();
-
-    const { output: closedOutput } = shallow(component.getByType(Modal));
-    expect(closedOutput).toMatchSnapshot();
-  });
+    fireEvent.press(component.getByTestId(buttonShowModalTestId));
+    fireEvent.press(component.getByTestId(buttonHideModalTestId));
 
-  it('* component styled with mappings', () => {
-    const component: RenderAPI = render(
-      <Modal visible={true}>
-        <Text>Test</Text>
-      </Modal>,
-    );
+    const modalInstance: any = component.getByType(Modal).instance;
 
-    const { output } = shallow(component.getByType(Modal));
+    const absoluteStyle: StyleType = modalInstance.getAbsoluteRelatedStyle();
+    const expectedStyle: StyleType = {
+      top: (height - modalInstance.contentHeight) / 2,
```

Comment:
you should check style props here instead of local variables

Reply:
Refactored.